grpc-proto/src/main/java/io/stargate/grpc/TypeSpecs.java

```diff
@@ -0,0 +1,458 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.grpc;
+
+import io.stargate.proto.QueryOuterClass.TypeSpec;
+import io.stargate.proto.QueryOuterClass.TypeSpec.List;
+import io.stargate.proto.QueryOuterClass.TypeSpec.Map;
+import io.stargate.proto.QueryOuterClass.TypeSpec.Set;
+import io.stargate.proto.QueryOuterClass.TypeSpec.Tuple;
+import io.stargate.proto.QueryOuterClass.TypeSpec.Udt;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class TypeSpecs {
```

Comment:
Temporarily cherry-picked from #1794 to avoid PRs on top of PRs. I'll remove the commit once #1794 gets merged.